types/mongodb/index.d.ts

```diff
@@ -585,9 +586,9 @@ export interface Collection<TSchema = Default> {
     rename(newName: string, callback: MongoCallback<Collection<TSchema>>): void;
     rename(newName: string, options?: { dropTarget?: boolean }): Promise<Collection<TSchema>>;
     rename(newName: string, options: { dropTarget?: boolean }, callback: MongoCallback<Collection<TSchema>>): void;
-    //http://mongodb.github.io/node-mongodb-native/2.1/api/Collection.html#replaceOne
+    //http://mongodb.github.io/node-mongodb-native/2.2/api/Collection.html#replaceOne
     replaceOne(filter: Object, doc: Object, callback: MongoCallback<UpdateWriteOpResult>): void;
-    replaceOne(filter: Object, doc: Object, options?: ReplaceOneOptions): Promise<UpdateWriteOpResult>;
+    replaceOne(filter: Object, doc: Object, options?: ReplaceOneOptions): Promise<UpdateWriteOpResult> | Promise<WriteOpResult>;
```

Comment:
I would change it with `Promise<UpdateWriteOpResult & { ops:Array<any> }>` since using the `|` it would require a cast to `UpdateWriteOpResult` to access its other property.  I've done a couple of test and it seems that the `ops` property is always there, regardless of the value of upsert.